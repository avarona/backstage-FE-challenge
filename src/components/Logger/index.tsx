import React from 'react';
import styles from './styles.module.scss';

const Logger = () => (
  <div className={styles.loggerContainer}>
    This is the logger.
  </div>
)

export default Logger;
