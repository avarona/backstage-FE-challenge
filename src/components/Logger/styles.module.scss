.loggerContainer {
  height: 40%;
  border-top: 1px solid #aaa;
  overflow-y: auto;
  list-style: none;
}

.loggerItem {
  padding: 8px 0;
}
