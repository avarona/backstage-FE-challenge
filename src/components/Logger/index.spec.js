import React from 'react';
import { shallow } from 'enzyme';
import Logger from './';

describe('Logger', () => {
  describe('with logs', () => {
    const logs = ['1', '2', '3'];
    const wrapper = shallow(<Logger log={logs} />);

    it('renders', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('has a LogItem', () => {
      expect(wrapper.find('LogItem').exists()).toBe(true);
    });
  });

  describe('with no logs', () => {
    const wrapper = shallow(<Logger log={[]} />);

    it('renders', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not have a LogItem', () => {
      expect(wrapper.find('LogItem').exists()).toBe(false);
    });
  });
});
