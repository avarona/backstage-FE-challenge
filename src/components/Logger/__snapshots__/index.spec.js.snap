// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger with logs renders 1`] = `"<ul class=\\"loggerContainer\\"><li class=\\"loggerItem\\"><ul class=\\"logItemContainer\\"><li><strong>0:</strong> 1</li></ul></li><li class=\\"loggerItem\\"><ul class=\\"logItemContainer\\"><li><strong>0:</strong> 2</li></ul></li><li class=\\"loggerItem\\"><ul class=\\"logItemContainer\\"><li><strong>0:</strong> 3</li></ul></li></ul>"`;

exports[`Logger with no logs renders 1`] = `"<ul class=\\"loggerContainer\\"></ul>"`;
